### ISSUE TITLE
Key without geolocation permissions fails geolocation requests with no indication why

### ISSUE THREAD

**Issue by stuartlangridge** [51Degrees/location-node#2]
If the location-node module is used with a key that does not have permission to make geolocation calls, it fails and there is no indication of what the failure is.

Below a key is used which causes this to fail.

```javascript
const FiftyOneDegreesGeoLocation = require('./index');
let localResourceKey = 'AQQNX4tozS1CfutY2Eg';
const pipeline = new FiftyOneDegreesGeoLocation.GeoLocationPipelineBuilder({
  resourceKey: localResourceKey
}).build();
pipeline.on('error', err => { console.error("woo", err)});

const getCountry = async function (latitude, longitude) {
  const flowData = pipeline.createFlowData();
  flowData.evidence.add('query.51D_Pos_latitude', latitude);
  flowData.evidence.add('query.51D_Pos_longitude', longitude);
  await flowData.process();
  console.log(flowData.location.country); // undefined
  console.log(flowData.location); // no errors are indicated
};

getCountry('51', '-1');
```

This library call essentially boils down to a call to the HTTP endpoint https://cloud.51degrees.com/api/v4/AQQNX4tozS1CfutY2Eg.json?51D_Pos_latitude=51&51D_Pos_longitude=-1

Note that the results indicate that geolocation is not enabled on this key (I think):

```sh
$ curl "https://cloud.51degrees.com/api/v4/AQQNX4tozS1CfutY2Eg.json?51D_Pos_latitude=51&51D_Pos_longitude=-1"
{
  "configuration-element": {},
  "access-control": {},
  "products": {},
  "global-tracker": {},
  "tracker": {},
  "accessible-properties": {},
  "location": {},
  "location_digitalelement": {},
  "device": {
    "geolocation": false
  },
  "hardware": {}
}
```

but this is not surfaced at all by the JavaScript library.

Note that the examples suggest checking `flowData.location.country.hasValue`, but `flowData.location.country` is undefined in this situation and so trying to read `flowData.location.country.hasValue` will throw an error.

I am not sure whether this is an issue with the library (should it work harder to expose these errors) or the API itself, or documentation. Certainly if it is expected that users can create keys which do not have the permissions required to query the geolocation API, that it needs to be obvious that they have done so, and how to get a new key which does have those permissions (https://configure.51degrees.com does not make this very clear at all!)

Since the library essentially builds its response from the server response (that is, the library itself doesn't "know" what an API call "should" return; it just generates a response based on what the server returns), it is possible that the best solution here is, when a geolocation API call is made with a key that is not authorised for geolocation calls, that the server should _still return_ geolocation data which is null and with `countrynullreason` set to something like "unauthorised key", which would make the problem clear for all consumers regardless of their library choice. (That is: this would mean that `flowdata.location.country` is defined and null and has a reason for failure, and so `flowData.location.country.hasValue` will also be defined and available to check, rather than not existing at all.)

**Comment by tungntpham** [comment]
HI @stuartlangridge . Many thanks. This will be addressed in the release v4.2.0.

**Comment by tungntpham** [comment]
Version 4.2.0 is now released.
